### PR TITLE
Misc

### PR DESCRIPTION
**Pull request by pascal-brand-st-dev**

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

File: public/tee_client_api.h [hunk 1]
```diff
@@ -275,6 +275,10 @@ typedef struct {
 	size_t size;
 	uint32_t flags;
 	/* Implementation-Defined, must match what the kernel driver have */
+	/*
```
Comment (jbech-linaro) [inline public/tee_client_api.h]: Combine this comment with previous line.

File: tee-supplicant/src/tee_supplicant.c [hunk 2]
```diff
@@ -409,9 +411,14 @@ int main(int argc, char *argv[])
 
 	while (true) {
 		struct tee_rpc_invoke request;
+		int ret;
 		DMSG("looping");
 
-		if (read_request(fd, &request)) {
+		/* major failure on read kills suppl, malformed data will not. */
```
Comment (jbech-linaro) [inline tee-supplicant/src/tee_supplicant.c]: *suppl*, please spell the complete name.

File: tee-supplicant/src/tee_supplicant.c [hunk 3]
```diff
@@ -409,9 +411,14 @@ int main(int argc, char *argv[])
...
+		ret = read_request(fd, &request);
+		if (ret < 0)
```
Comment (jbech-linaro) [inline tee-supplicant/src/tee_supplicant.c]: What about changing from:
```
while (true) {
int ret;
...
}
```
to
```
int ret = 0;
...
while (ret >= 0) {
...
}
```

File: tee-supplicant/src/tee_supplicant.c [hunk 4]
```diff
@@ -449,28 +456,31 @@ int main(int argc, char *argv[])
...
+
+	if ((size_t)res < sizeof(*request) - sizeof(request->cmds)) {
 		EMSG("error reading from driver");
-		return false;
+		return 1;
 	}
 
 	if (sizeof(*request) - sizeof(request->cmds) +
-	    sizeof(request->cmds[0]) * request->nbr_bf != res) {
+	    sizeof(request->cmds[0]) * request->nbr_bf != (size_t)res) {
```
Comment (jbech-linaro) [inline tee-supplicant/src/tee_supplicant.c]: `size_t` or `ssize_t`?

File: tee-supplicant/src/tee_supplicant.c (hunk 3 above)
Comment (pascal-brand-st-dev) [inline tee-supplicant/src/tee_supplicant.c]: In that case, what about

    int ret;
    ...
    do {
    ...
    } while (ret >= 0);

Comment (jbech-linaro) [inline tee-supplicant/src/tee_supplicant.c]: Even better!

File: tee-supplicant/src/tee_supplicant.c (hunk 4 above)
Comment (pascal-brand-st-dev) [inline tee-supplicant/src/tee_supplicant.c]: I think size_t is correct.

File: tee-supplicant/src/tee_supplicant.c (hunk 2 above)
Comment (pascal-brand-st-dev) [inline tee-supplicant/src/tee_supplicant.c]: ok

File: public/tee_client_api.h (hunk 1 above)
Comment (pascal-brand-st-dev) [inline public/tee_client_api.h]: ok

**Comment by pascal-brand-st-dev** [comment]
Rebase and tag applied.

**Pull request merged**